tutorial.md

```diff
@@ -0,0 +1,131 @@
+                              text="products/compute/api.yaml">
+</walkthrough-editor-open-file>.
+
+### Product metadata
+
+The
+<walkthrough-editor-select-regex filePath="magic-modules/products/compute/api.yaml"
+                                 regex="!ruby/object:Api::Product"
+                                 text="top section">
+</walkthrough-editor-select-regex>
+provides metadata about the API, such as name, scopes, and versions.
+
+### Resources
+
+Each `api.yaml` file contains a list of resources.
+Let's
+<walkthrough-editor-select-regex filePath="magic-modules/products/compute/api.yaml"
+                                 regex="!ruby/object:Api::Resource"
+                                 text="look at">
+</walkthrough-editor-select-regex>
+the first one.
+
+This section contains data about the resource, such as its name, description, and URLs.
+
+### Parameters + Properties
+
+Each resource contains a list of
+<walkthrough-editor-select-regex filePath="magic-modules/products/compute/api.yaml"
+                                 regex="parameters:"
+                                 text="URL parameters">
+</walkthrough-editor-select-regex>
+and
+<walkthrough-editor-select-regex filePath="magic-modules/products/compute/api.yaml"
+                                 regex="properties:"
+                                 text="properties">
+</walkthrough-editor-select-regex>
+on the resource that a user might set when creating the resource, or access when reading it.
+
+See the
+<walkthrough-editor-open-file filePath="magic-modules/DEVELOPER.md"
+                              text="Developer Guide">
+</walkthrough-editor-open-file>
+for more information about the values that can be set on parameters/properties.
+
+## [provider].yaml
+
+Within each product directory, each provider has its own `[provider].yaml` file to set information
+specific to that provider.
+
+Let's look at
+<walkthrough-editor-open-file filePath="magic-modules/products/compute/terraform.yaml"
+                              text="products/compute/terraform.yaml">
+</walkthrough-editor-open-file>.
+
+This file consists of information that is specific to Terraform, such as examples, modified
+descriptions, custom code, and validation functions.
+
+## Making changes
+
+To add a new API or resource, the only files that need to be modified are `api.yaml`, each
+`[provider].yaml`, and any custom code or provider-specific extras (such as Terraform example templates).
+
+Let's actually make a change. Go back to
+<walkthrough-editor-open-file filePath="magic-modules/products/compute/api.yaml"
+                              text="products/compute/api.yaml">
+</walkthrough-editor-open-file>
+and change the description on the `Address` resource.
+
+## Compiling magic-modules
+
+Now, let's compile those changes.
+
+Since we're running in cloud shell, this command will make sure we connect to GitHub via HTTPS
```

Comment:
Maybe move this Ansible configuration to its own subsection? That way, it's even more obvious that this isn't a part of the normal workflow

Reply:
Done